### DISCUSSION TITLE
Unit question

### DISCUSSION THREAD

**Discussion by rebeltroopers** [brucefan1983/GPUMD#737] [Q&A] [answered]
Hello, I am doing work on Polymers and am trying to get physical properties of them. For the stress output are they in units stated are F/A which would lead it to be eV/A3. Is this assumption correct or is it converted to GPa like in the thermo.out. I tried looking for a while and could not find it, or missed it.

Thanks for you time.

**Comment by brucefan1983** [comment]
Sorry I forgot to answer this question. But I am not clear what "output" you refer to. Is it the output in `dump.xyz`?

**Reply by rebeltroopers** [reply]
Sorry been very busy. yes I mean the output dump.xyz. From my calculations it seems to be in eV/A.

Also one more question, my system is very sensitive to the pairwise cutoffs. I know that the new use_typewise_cutoff was created. I need to limit H to be much shorter than my normal cutoff. How do you know what the covalent distance is as you describe in your section? Can I set it manually or is it guessed like H-H (0.7A) distance or C-H(1.1A) distance? Do you happen to have an example as well somewhere? This seems to affect my vdW heavily.

**Reply by brucefan1983** [reply]
OK, for the `dump.xyz` file, the stress is in units of eV/A^3.

The covalent radii for all supported elements are written here:
https://github.com/brucefan1983/GPUMD/blob/master/src/utilities/nep_utilities.cuh

I reproduce them here:
```c++
__constant__ float COVALENT_RADIUS[94] = {
  0.426667f, 0.613333f, 1.6f,     1.25333f, 1.02667f, 1.0f,     0.946667f, 0.84f,    0.853333f,
  0.893333f, 1.86667f,  1.66667f, 1.50667f, 1.38667f, 1.46667f, 1.36f,     1.32f,    1.28f,
  2.34667f,  2.05333f,  1.77333f, 1.62667f, 1.61333f, 1.46667f, 1.42667f,  1.38667f, 1.33333f,
  1.32f,     1.34667f,  1.45333f, 1.49333f, 1.45333f, 1.53333f, 1.46667f,  1.52f,    1.56f,
  2.52f,     2.22667f,  1.96f,    1.85333f, 1.76f,    1.65333f, 1.53333f,  1.50667f, 1.50667f,
  1.44f,     1.53333f,  1.64f,    1.70667f, 1.68f,    1.68f,    1.64f,     1.76f,    1.74667f,
  2.78667f,  2.34667f,  2.16f,    1.96f,    2.10667f, 2.09333f, 2.08f,     2.06667f, 2.01333f,
  2.02667f,  2.01333f,  2.0f,     1.98667f, 1.98667f, 1.97333f, 2.04f,     1.94667f, 1.82667f,
  1.74667f,  1.64f,     1.57333f, 1.54667f, 1.48f,    1.49333f, 1.50667f,  1.76f,    1.73333f,
  1.73333f,  1.81333f,  1.74667f, 1.84f,    1.89333f, 2.68f,    2.41333f,  2.22667f, 2.10667f,
  2.02667f,  2.04f,     2.05333f, 2.06667f};
```

This table is taken from the the DFT-D3 approach, and the data are in units of A here.

For H-H pairs, if you use `use_typewise_cutoff 2.5 2`, the radial cutoff radius will be 0.426667 A * 2 * 2.5 = 2.133335 A, and the angular cutoff radius will be 0.426667 A * 2 * 2.5 =1.706668 A, if your global cutoff radii are larger than these. 

Currently, you can tune the factors for the `use_typewise_cutoff` keyword, but cannot set the cutoff radius for each pair directly (that would take a lot of inputs for a big model such as one for the periodic table).

**Reply by rebeltroopers** [reply]
TY for your detailed information. You have answered it very well.

Just one last confirmation that the C-H would be (.426667 + 1.0)*2.5 = 2.84A with your given scheme for the radial term.
My reactive chemistry is affecting the H-H and C-H bonds a little to much, so I need to cut them down a bit.

**Reply by brucefan1983** [reply, marked as answer]
Yes, the CH cutoff is calculated in the way your presented. Hope this typewise cutoff scheme is effective in your problem.
